src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -29,6 +29,127 @@ class OrchestratorCli(orchestrator.OrchestratorClientMixin, MgrModule):
+            }
+        self.set_health_checks(h)
+
+    def _get_device_hosts(self, dev_id):
+        locs = [d['location'] for d in self.get('devices')['devices'] if d['devid'] == dev_id]
+        return [l['host'] for loc in locs for l in loc]
+
+    @CLIReadCommand(prefix='device ls-lights',
+                    desc='List currently active device indicator lights')
+    def _command_ls(self):
+        return HandleCommandResult(
+            stdout=json.dumps({
+                'ident': list(self.ident),
+                'fault': list(self.fault)
+                }, indent=4))
+
+    @CLIWriteCommand(prefix='device fault-light-on',
+                     args='name=devid,type=CephString',
+                     desc='Enable device *fault* light')
+    def _command_fault_on(self, devid):
+        hosts = self._get_device_hosts(devid)
+        if not hosts:
+            return HandleCommandResult(stderr='device %s not found' % devid,
+                                       retval=-errno.ENOENT)
+        self.fault.add(devid)
+        self._save()
+        self._refresh_health()
+        completions = [self.blink_device_light("fault", True, h, devid) for h in hosts]
+        self._orchestrator_wait(completions)
+        return HandleCommandResult(stdout='\n'.join(c.result for c in completions))
+
+    @CLIWriteCommand(prefix='device ident-light-on',
+                     args='name=devid,type=CephString',
+                     desc='Enable device *ident* light')
+    def _command_ident_on(self, devid):
+        hosts = self._get_device_hosts(devid)
+        if not hosts:
+            return HandleCommandResult(stderr='device %s not found' % devid,
+                                       retval=-errno.ENOENT)
+        self.ident.add(devid)
+        self._save()
+        self._refresh_health()
+        completions = [self.blink_device_light("ident", True, h, devid) for h in hosts]
+        self._orchestrator_wait(completions)
+        return HandleCommandResult(stdout='\n'.join(c.result for c in completions))
+
+
+    @CLIWriteCommand(prefix='device fault-light-off',
+                     args='name=devid,type=CephString',
+                     desc='Disable device *fault* light')
+    def _command_fault_off(self, devid):
+        hosts = self._get_device_hosts(devid)
+        if not hosts:
+            return HandleCommandResult(stderr='device %s not found' % devid,
+                                       retval=-errno.ENOENT)
+
+        completions = [self.blink_device_light("fault", False, h, devid) for h in hosts]
+        self._orchestrator_wait(completions)
+
+        if devid in self.fault:
+            self.fault.remove(devid)
+            self._save()
+            self._refresh_health()
+        return HandleCommandResult(stdout='\n'.join(c.result for c in completions))
+
+    @CLIWriteCommand(prefix='device ident-light-off',
+                     args='name=devid,type=CephString',
```

Comment:
I think for *-off we also need a --force-cleanup flag that skips the orchestrator callout and just cleans up the self.ident or self.fault list.  that way we can make the cluster shut up about a device on a host that is no longer reachable.

Reply:
we just have to make sure, the flag is cleared in case of a broken host, right?

Reply:
Well.. I would lean towards a default behavior of erroring out and not removing the device from the list when the host in unreachable.  Otherwise a network or other issue might make the off command fail but we'd still remove the device from the set, leaving the light on.  Better to have the user (or consuming layer) explicitly ask to clear the light state without doing the lsmcli command.